packages/nft-minter/src/mint.ts

```diff
@@ -0,0 +1,129 @@
+import { EventEmitter } from "events";
+import {
+  Address,
+  Chain,
+  createPublicClient,
+  Hash,
+  http,
+  TransactionReceipt,
+  WalletClient,
+} from "viem";
+import {
+  getBalance,
+  readContract,
+  waitForTransactionReceipt,
+  writeContract,
+} from "viem/actions";
+
+import { erc721Abi } from "./abi";
+
+type Events = {
+  "minting-failed": [Error];
+  "minting-failed-validation": [string];
+  "minting-transaction-succeeded": [TransactionReceipt];
+  "minting-transaction-reverted": [TransactionReceipt];
+  "pre-mint": [];
+  "unexpected-error": [Error];
+  "user-signed-approved": [Hash];
+  "user-signing-error": [Error];
+};
+
+const toPromiseEvent = function (
+  callback: (emitter: EventEmitter<Events>) => Promise<void>,
+) {
+  const emitter = new EventEmitter<Events>();
+
+  // eslint-disable-next-line promise/no-callback-in-promise
+  const promise = Promise.resolve().then(() => callback(emitter));
```

Comment:
What about simplifying this as follows?

```suggestion
  const promise = Promise.resolve(callback(emitter));
```

Reply:
Improved here [606318f](https://github.com/hemilabs/tge-mint-page/pull/2/commits/606318f41f93b7e94f39371ced30224d64335f2f)